okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -199,18 +199,30 @@ class HttpLoggingInterceptor @JvmOverloads constructor(
       } else if (requestBody.isOneShot()) {
         logger.log("--> END ${request.method} (one-shot body omitted)")
       } else {
-        val buffer = Buffer()
+        var buffer = Buffer()
         requestBody.writeTo(buffer)
 
+        var gzippedLength: Long? = null
+        if ("gzip".equals(headers["Content-Encoding"], ignoreCase = true)) {
+          gzippedLength = buffer.size
+          GzipSource(buffer.clone()).use { gzippedResponseBody ->
```

Comment:
Is the clone() necessary?

Reply:
Good point, copied from the response handling.